spec.html

```diff
@@ -34894,7 +34894,7 @@ <h1>String.prototype.match ( _regexp_ )</h1>
 
       <emu-clause id="sec-string.prototype.matchall">
         <h1>String.prototype.matchAll ( _regexp_ )</h1>
-        <p>This method performs a regular expression match of the String representing the *this* value against _regexp_ and returns an iterator. Each iteration result's value is an Array containing the results of the match, or *null* if the String did not match.</p>
+        <p>This method performs a regular expression match of the String representing the *this* value against _regexp_ and returns an iterator. Each iteration result's value is an Array containing the results of the match.</p>
```

Comment:
This is fine, but I would probably say a bit more.

```suggestion
        <p>This method performs a regular expression match of the String representing the *this* value against _regexp_ and returns an iterator that yields match results. Each match result is an Array containing the matched portion of the String as the first element, followed by the portions matched by any capturing groups. If the regular expression never matches, the returned iterator does not yield any match results.</p>
```

Reply:
We could also `<dfn>` a term for these match result arrays constructed and returned by `RegExpBuiltinExec`. /cc @tc39/ecma262-editors

Reply:
In the editor call today, @syg agreed that we should probably `<dfn>` this thing. We could do it as part of this PR or as a follow-up, doesn't matter to me.

Reply:
Thank you for the review!

I'm totally not sure how to add a definition for the term, so I'd prefer not to address that in this PR.